buffer.go

```diff
@@ -36,61 +44,74 @@ func (b *Buffer) HandleMeasures(time time.Time, measures ...Measure) {
 		return
 	}
 
-	size := b.BufferSize
-	if size == 0 {
-		size = 1024
-	}
-
-	chunk := acquireChunk()
-	chunk.reset()
-	chunk.append(b.Serializer, time, measures...)
-
-	// Chunks that are already larger than the maximum buffer size are directly
-	// passed to the serializer, no need to aggregate them in a larger buffer.
-	if chunk.len() >= size {
-		chunk.writeTo(b.Serializer)
-	} else {
-		for {
-			var flush *buffer
+	size := b.bufferSize()
+	b.init(size)
 
-			b.mutex.Lock()
+	for {
+		buffer := b.acquireBuffer()
+		length := buffer.len()
+		buffer.append(b.Serializer, time, measures...)
 
-			if b.buffer == nil {
-				b.buffer = acquireBuffer()
-				b.buffer.reset(size)
-			}
+		if buffer.len() < size {
+			buffer.release()
+			break
+		}
 
-			if (b.buffer.len() + chunk.len()) > b.buffer.cap() {
-				b.buffer, flush = nil, b.buffer
-			} else {
-				b.buffer.append(chunk)
-			}
+		if length != 0 {
```

Comment:
I don't understand what this is doing.

If we get `length` before updating `buffer.data`  on line 53, couldn't we be potentially truncating things off here?  I would get truncating to `size` but I'm sure I'm just missing something.

Reply:
It prevents from exceeding the buffer size in the common case, when the buffer is full it’ll get flushed and the code loops back to write the measures. 
Now that I’m re-reading I realize this code would likely cause an infinite loop since we’ll keep retrying, thanks for pointing it out (seems like this is calling for a test).

Reply:
I thought the break statement above would take care of not exceeding the buffer size.

Totally missed the infinite loop after write issue though.  I think I understood what you were getting at in my head and missed that.